vhdbuilder/packer/generate-vhd-publishing-info.sh

```diff
@@ -81,35 +80,27 @@ else
     IMAGE_ARCH="x64"
 fi
 
-echo "generating traditional SAS token with CLASSIC_SA_CONNECTION_STRING..."
-start_date=$(date +"%Y-%m-%dT00:00Z" -d "-1 day")
-expiry_date=$(date +"%Y-%m-%dT00:00Z" -d "+1 year")
-if [[ "${OS_NAME,,}" != "windows" ]]; then
-    [ -z "${OUTPUT_STORAGE_CONTAINER_NAME}" ] && echo "OUTPUT_STORAGE_CONTAINER_NAME should be set..." && exit 1
-    echo "storage container name: ${OUTPUT_STORAGE_CONTAINER_NAME}"
-    # max of 7 day expiration time when using user delegation SAS
-    sas_token=$(az storage container generate-sas --name ${OUTPUT_STORAGE_CONTAINER_NAME} --permissions lr --connection-string ${CLASSIC_SA_CONNECTION_STRING} --start ${start_date} --expiry ${expiry_date} | tr -d '"')
-else
-    # we still need to use the original connection string when not using a system-assigned identity on 1ES pools
-    sas_token=$(az storage container generate-sas --name vhds --permissions lr --connection-string ${CLASSIC_SA_CONNECTION_STRING} --start ${start_date} --expiry ${expiry_date} | tr -d '"')
-fi
+# It will automatically generate the user delegation sas token in publishing the image so it does not need to store sas token in vhd-publishing-info.json
+vhd_url="${STORAGE_ACCT_BLOB_URL}/${VHD_NAME}"
+
+echo "generating user delegation SAS token to validate ${vhd_url}"
+vhd_url_storage_account_fqdn=$(echo ${vhd_url}| cut -d "/" -f 3)
+vhd_url_storage_account=$(echo ${vhd_url_storage_account_fqdn}| cut -d "." -f 1)
+vhd_url_storage_container=$(echo ${vhd_url}| cut -d "/" -f 4)
+expiry_date=$(date +"%Y-%m-%dT00:00Z" -d "+2 day")
+sas_token=$(az storage container generate-sas --account-name ${vhd_url_storage_account} --name ${vhd_url_storage_container} --permissions r --expiry ${expiry_date} --auth-mode login --as-user | tr -d '"')
```

Comment:
not sure there's much value in testing the user-delegation SAS token here anymore since the underlying identity that generates the SAS token during release-time will be different than the identity on the 1ES pool used for running these builds

Reply:
It is used to check whether the vhd is there but not validate the sas token.

Reply:
This also can validate whether the msi has the correct roles to the storage account.

Reply:
Do not generate user-delegation sas token.